src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java

```diff
@@ -172,20 +174,17 @@ private TransactionClose(String msg) {
     }
 
     public List<KeyValue> getRange(final byte[] startKey, final byte[] endKey,
-                                            final int limit) throws PermanentBackendException {
+                                   final int limit) throws PermanentBackendException {
         boolean failing = true;
         List<KeyValue> result = Collections.emptyList();
         for (int i = 0; i < maxRuns; i++) {
-            final int startTxId = txCtr.get();
+            txCtr.get();
             try {
-                result = tx.getRange(new Range(startKey, endKey), limit).asList().get();
+                ReadTransaction transaction = getTransaction(isolationLevel, this.tx);
+                result = transaction.getRange(new Range(startKey, endKey), limit).asList().join();
```

Comment:
get() throws interrupted exception. was blocked on figuring out why interruptException is happening. while join doesn't throws interrupt exception instead throws RuntimeException. Hoping for the best.